src/ensembl/src/services/storage-service.ts

```diff
@@ -75,7 +82,7 @@ export class StorageService implements StorageServiceInterface {
   public update(key: string, fragment: JSONValue, options = defaultOptions) {
     const storedData = this.get(key, options);
     if (storedData) {
-      this.save(key, merge(storedData, fragment), options);
+      this.save(key, mergeWith(storedData, fragment, overwriteArray), options);
```

Comment:
You might want to add a test for this [here](https://github.com/Ensembl/ensembl-client/blob/dev/src/ensembl/src/services/tests/storage-service.test.ts#L100-L125)

Reply:
That's done